var/spack/repos/builtin/packages/mcl/package.py

```diff
@@ -15,3 +15,13 @@ class Mcl(AutotoolsPackage):
     url      = "https://www.micans.org/mcl/src/mcl-14-137.tar.gz"
 
     version('14-137', 'bc8740456cf51019d0a9ac5eba665bb5')
+
+    depends_on('perl')
+
+    def configure_args(self):
+        return ['--enable-blast']
+
+    @run_after('build')
+    def fix_perl_scripts(self):
+        filter_file(r'#!/usr/local/bin/perl -w',
+                    '#!/usr/bin/env perl', prefix.bin.mcxdeblast)
```

Comment:
It would be better to wire the script up to use the perl that you're depending on above.

E.g. something like ([from the vcf repo](https://github.com/spack/spack/blob/1e1ea54bada54ae783198823c7f8fe0430ac41cb/var/spack/repos/builtin/packages/vcftools/package.py#L31-L52)):

```
"#!{perl}".format(perl=self.spec['perl'].command.path)
```

Reply:
looks good.  I haven't tried any of the other bits.